moler/device/textualdevice.py

```diff
@@ -241,6 +262,32 @@ def name(self):
     def name(self, value):
         self._name = value
 
+    @property
+    def public_name(self):
+        """
+        Getter for publicly used device name.
+
+        Internal name of device (.name attribute) may be modified by device itself  in some circumstances (to not
+        overwrite logs). However, public_name is guaranteed to be preserved as it was set by external/client code.
+
+        :return: String with the device alias name.
+        """
+        ret = self.name
+        if self._public_name:
+            ret = self._public_name
+        return ret
+
+    @public_name.setter
+    def public_name(self, value):
```

Comment:
should we ever allow for changing `public_name` or it is "name of creation time"

But if we allow then we will get discrepancy between logs naming and current name of device in memory. Maybe we should rename logs using the same machinery as we have used during device creation.

Reply:
Can be set only once.